server/src/models/pmcmean.rs

```diff
@@ -87,61 +85,57 @@ impl PMCMean {
     fn is_value_within_error_bound(&self, real_value: Value, approximate_value: Value) -> bool {
         // Needed because result becomes NAN and approximate_value is rejected
         // if approximate_value and real_value are zero, and because NAN != NAN.
-        if equal_or_nan(real_value, approximate_value) {
-            true
-        } else {
-            let difference = real_value - approximate_value;
-            let result = Value::abs(difference / real_value);
-            (result * 100.0) <= self.error_bound
-        }
+	if models::equal_or_nan(real_value as f64, approximate_value as f64) {
```

Comment:
I do not think it will be an issue. However, converting a `f32` to a `f64` is performed using a `cvtss2sd` instruction on x86_64 so unsurprisingly both [Rustc](https://godbolt.org/z/cjP8Y3P1h) and [Clang](https://godbolt.org/z/f37M54h95) add two `cvtss2sd` instructions. Technically we can make [Rustc](https://godbolt.org/z/seKK33vd9) and [Clang](https://godbolt.org/z/19PjP86Kc) remove one of them by only casting the result. However, as we then would need two instances of the function to handle parameters as `f32` and `f64` we may as well make the function generic and let the compiler generate the both versions using [monomorphization](https://rustc-dev-guide.rust-lang.org/backend/monomorph.html).

Reply:
I agree.